### PR TITLE
Add MonthPicker 

### LINKED ISSUE #202: Add MonthPicker to DatePicker
### Description

Extend the DatePicker component and enable MonthPicker mode.
A beautiful example is the DatePicker component of Buefy and since Oruga inherits from Buefy it should be easy to implement 🐛

https://buefy.org/documentation/datepicker#month-picker

### Why Oruga need this feature

It's a component that's widely used on websites and libraries like Buefy and Vuetify already have it.

### PR DESCRIPTION
**Pull request by mirkoperillo**
Fixes #202 

Work in progress.
For the moment added `DatepickerMonth` component from https://github.com/buefy/buefy/blob/dev/src/components/datepicker/DatepickerMonth.vue

Need to fix some tests that currently fail

## Proposed Changes

- add `DatepickerMonth.vue` and ` DatepickerMonth.spec.js` to datepicker folder

### CODE REVIEW

**Comment by astagi** [comment]
@mirkoperillo your work looks good so far! any updates here?

**Comment by mirkoperillo** [comment]
@astagi I miss a couple of thing:
1. I have one problem, 2 unit test of `DatepickerMonth.spec.js` fail (and I don't understand why because are the same of `buefy`).
2. I'm working on the refactor of `DatepickerMonth` component to introduce `rowClass` function and the others taking `DatepickerTable` as sample.

**Comment by mirkoperillo** [comment]
Disclaimer: I have two doubts but I need some mentoring to improve this PR, so I decided to declare it ready for the review:

Doubts:
1. Two unit tests inherited from the buefy component fail and I'm not understand why (currently put in comments)

https://github.com/mirkoperillo/oruga/blob/ece9354b91bf6249d0076b31baaeb728c5992bbe/packages/oruga/src/components/datepicker/DatepickerMonth.spec.js#L353

and this

https://github.com/mirkoperillo/oruga/blob/ece9354b91bf6249d0076b31baaeb728c5992bbe/packages/oruga/src/components/datepicker/DatepickerMonth.spec.js#L431

2. I'm not sure if the requested refactor of the component following the `DatepickerTable` component can be considered complete or not.

Waiting for your hints.

**Comment by jtommy** [comment]
@mirkoperillo 

> Two unit tests inherited from the buefy component fail and I'm not understand why (currently put in comments)

Don't worry about, we'll take a look to understand the problem

> I'm not sure if the requested refactor of the component following the DatepickerTable component can be considered complete or not.

Nothing has been change since your last PR.. Looking the code there is already duplicated code (for example `selectableDate`, `manageKeydown`, ... methods) and the component contains static class (for example `classObject` method).
Did you see https://github.com/oruga-ui/oruga/blob/develop/packages/oruga/src/components/datepicker/DatepickerTableRow.vue component? You might reuse the same code on both components.

**Comment by mirkoperillo** [comment]
@jtommy I'll give a look to your hint.

Maybe I've not understand what do you mean for this refactor, in the last commit I have introduced something like this
```
 <section :class="tableClasses">
```

instead of

```
<section class="datepicker-table">
```

**Comment by jtommy** [comment]
Yes but only in the template

**Comment by mirkoperillo** [comment]
@jtommy  do you think extracting the common methods (like `selectableDate`, `manageKeydown` and others) in an helper or a mixin file could be a good idea ?

**Comment by jtommy** [comment]
Sure

**Comment by mirkoperillo** [comment]
Thank you @jtommy 
Currently I'm still working on it.

I have a lot of problems with the introduction of `cellClasses` method instead of the `classObject` one.
A lot of tests are broken because it seems any expected classes is correctly assigned to the elements of the component.

**Comment by jtommy** [comment]
Ok but go ahead and I'll help you with tests

**Comment by mirkoperillo** [comment]
Ok, tomorrow I think I'll push the stuff I'm working on

**Comment by mirkoperillo** [comment]
@jtommy I have just pushed the last stuff on PR.
As I said yesterday, currently I have problems with the unit tests

**Comment by jtommy** [comment]
@mirkoperillo in the meanwhile, can you resolve and fix the conflicts?

**Comment by mirkoperillo** [comment]
sure

**Comment by mirkoperillo** [comment]
done

File: packages/oruga/src/components/datepicker/Datepicker.vue [hunk 1]
```diff
@@ -119,59 +119,83 @@
...
-                    :table-event-indicators-class="tableEventIndicatorsClass"
-                    @range-start="date => $emit('range-start', date)"
-                    @range-end="date => $emit('range-end', date)"
-                    @close="togglePicker(false)"
-                    @update:focused="focusedDateData = $event" />
-
-
+                <div
+                    v-if="!isTypeMonth"
+                    class="datepicker-content"
```
Comment (jtommy) [inline packages/oruga/src/components/datepicker/Datepicker.vue:124]: Why did you add this and the following lines?
You shouldn't touch the current use of `o-datepicker-table`

File: packages/oruga/src/components/datepicker/DatepickerMonth.vue [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+        focused: Object,
+        disabled: Boolean,
+        dateCreator: Function,
+        unselectableDates: [Array, Function],
+        unselectableDaysOfWeek: Array,
+        selectableDates: [Array, Function],
+        range: Boolean,
+        multiple: Boolean,
+        monthClass: [String, Function, Array],
+        monthBodyClass: [String, Function, Array],
```
Comment (jtommy) [inline packages/oruga/src/components/datepicker/DatepickerMonth.vue:66]: You added new *class props but you didn't add the class implementation into .scss, why?
Also, you should add the same props to Datepicker in order to pass them from datepicker instance

File: packages/oruga/src/components/datepicker/DatepickerMonth.vue [hunk 3]
```diff
@@ -0,0 +1,467 @@
+<template>
+    <section :class="monthClasses">
+        <div :class="monthBodyClasses">
+            <div class="datepicker-months">
```
Comment (jtommy) [inline packages/oruga/src/components/datepicker/DatepickerMonth.vue:4]: Why?

**Review by jtommy — CHANGES_REQUESTED** [review]

File: packages/oruga/src/components/datepicker/Datepicker.vue (hunk 1 above)
Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/Datepicker.vue:124]: I thought that the visibility of `o-datepicker-month` component or the `o-datapicker-table` one should be regulated by the value of `isTypeMonth` field. I used the same pattern used by `Datepicker` component of the `buefy` project

What the right thing to do ?

File: packages/oruga/src/components/datepicker/DatepickerMonth.vue (hunk 2 above)
Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:66]: I miss this thing, I'll try to fix it

File: packages/oruga/src/components/datepicker/DatepickerMonth.vue (hunk 3 above)
Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:4]: I have tried to transpose this thing looking the oruga `DatepickerTable` and the buefy one

File: packages/oruga/src/components/datepicker/Datepicker.vue (hunk 1 above)
Comment (jtommy) [reply packages/oruga/src/components/datepicker/Datepicker.vue:124]: Yes but we don't need that div, you can use `v-if` ok component itself or `template`

File: packages/oruga/src/components/datepicker/DatepickerMonth.vue (hunk 3 above)
Comment (jtommy) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:4]: I mean the third div static class

Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:4]: I have to fix it

File: packages/oruga/src/components/datepicker/Datepicker.vue (hunk 1 above)
Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/Datepicker.vue]: done

File: packages/oruga/src/components/datepicker/DatepickerMonth.vue (hunk 2 above)
Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:66]: done

File: packages/oruga/src/components/datepicker/DatepickerMonth.vue (hunk 3 above)
Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue]: done

File: packages/oruga/src/components/datepicker/DatepickerMonth.vue (hunk 2 above)
Comment (jtommy) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:64]: image

Probably there are other missing classes to apply

Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:64]: @jtommy how can I test visually the component (like you done for creating this screenshot) ?

**Comment by jtommy** [comment]
Running the docs, check contributing file with all instructions

File: packages/oruga/src/components/datepicker/DatepickerMonth.vue (hunk 2 above)
Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:66]: fixed the missing style class.
The layout is not still correct, but the problem is that the cells are not populated with `o-dpck__month_cell` class and this is a problem related with the failure of the unit tests
Screenshot 2021-11-23 at 10-27-34 UI components for Vue js and CSS framework agnostic
.

Comment (jtommy) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:66]: Why?

Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:66]: There isn't any space between months labels because all the links representing the months have the class attribute empty.
If they would have the `o-dpck__month__cell` the layout would be correct.
I'll try to do some rude debug to understand what happens

Comment (mirkoperillo) [reply packages/oruga/src/components/datepicker/DatepickerMonth.vue:66]: Fixed the style problem and now the tests are ok (except the two in comments)

**Comment by jtommy** [comment]
Thanks, you should add next version too so are you going to add it or we can work on it using your current code?

**Review by jtommy — APPROVED** [review]

**Comment by mirkoperillo** [comment]
@jtommy I can try to add this stuff to `oruga-next` too. I will ask some help if I would have problems

**Comment by jtommy** [comment]
@mirkoperillo keep in mind to change the following things:

- prop `value `-> modelValue
- event: `input `-> modelValue
- add `emits `array with all events thrown by datepicker
- remove `native `modifier

**Comment by mirkoperillo** [comment]
@jtommy I need your check, I think to have done

**Comment by jtommy** [comment]
@mirkoperillo thanks, I'll review as soon as possible
